features/deep-linking/conversation-navigator.ts

```diff
@@ -0,0 +1,105 @@
+import { useCallback } from "react"
+import { useNavigation } from "@react-navigation/native"
+import { captureError } from "@/utils/capture-error"
+import { IXmtpInboxId, IXmtpConversationId } from "@/features/xmtp/xmtp.types"
+import { logger } from "@/utils/logger"
+import { checkConversationExists } from "./conversation-links"
+
+/**
+ * Custom hook to handle conversation deep links
+ * This is used by the DeepLinkHandler component to navigate to conversations from deep links
+ */
+export function useConversationDeepLinkHandler() {
+  const navigation = useNavigation()
+  
+  /**
+   * Process an inbox ID from a deep link and navigate to the appropriate conversation
+   * @param inboxId The inbox ID from the deep link
+   * @param composerTextPrefill Optional text to prefill in the composer
+   */
+  const handleConversationDeepLink = useCallback(async (
+    inboxId: IXmtpInboxId, 
+    composerTextPrefill?: string
+  ) => {
+    if (!inboxId) {
+      logger.warn("Cannot handle conversation deep link - missing inboxId")
+      return
+    }
+
+    try {
+      logger.info(`Handling conversation deep link for inboxId: ${inboxId}${composerTextPrefill ? ' with prefill text' : ''}`)
+      
+      // Check if the conversation exists
+      const { exists, conversationId } = await checkConversationExists(inboxId)
```

Comment:
let's just use `findConversationByInboxIds`

Reply:
and if it returns undefined then we know we don't have any conversation for that inboxId